cpp/arcticdb/codec/codec.cpp

```diff
@@ -348,13 +348,14 @@ void decode_v2(
         const auto fields_size = desc.fields().size();
         const auto start_row = res.row_count();
         EncodedFieldCollection encoded_fields(std::move(encoded_fields_buffer));
-        const auto seg_row_count = fields_size ? ssize_t(encoded_fields.at(0).ndarray().items_count()) : 0L;
+        auto seg_row_count = fields_size ? ssize_t(encoded_fields.at(0).ndarray().items_count()) : ssize_t(0);
         res.init_column_map();
 
         for (std::size_t i = 0; i < fields_size; ++i) {
             const auto& encoded_field = encoded_fields.at(i);
             const auto& field_name = desc.fields(i).name();
             util::check(data!=end, "Reached end of input block with {} fields to decode", fields_size-i);
+            seg_row_count = std::max(seg_row_count, ssize_t(encoded_field.ndarray().items_count()));
```

Comment:
I assume `items_count` is the number of values actually stored, ignoring sparsity? If so, this will work right now, but there is a potential future use case where we have sparse data with a row-count index, where the segment row count depends on the column sparse maps as well. e.g.
```
lib.write(sym, pd.DataFrame("col1": [np.nan, 0.0], "col2": [1.0, np.nan]), sparsify_floats=True)
```
would set `seg_row_count` to `1` here, when it should be `2`. Please add a comment detailing this.

Reply:
Hmm, it's an interesting question. There is also EncodedField::values_count_ which I believe holds the value you're describing, but I'm not 100% sure. @willdealtry can you shed some light on this?

Reply:
I've discussed this with William, and we can actually do this properly now. After the iteration over fields, the result segment in memory has the columns and their sparse maps fully populated. Then we can just take the max over the last logical rows of all the columns to get the segment row count.